src/libcamera/meson.build

```diff
@@ -71,8 +71,11 @@ subdir('pipeline')
 subdir('proxy')
 
 libatomic = cc.find_library('atomic', required : false)
-libdl = cc.find_library('dl')
-libgnutls = cc.find_library('gnutls', required : true)
+libdl = dependency('', required : false)
+if not cc.has_function('dlopen')
+    libdl = cc.find_library('dl')
```

Comment:
does this need to be required : true  ?

Reply:
Perhaps that's default. I can't recall.

Reply:
mesa3d meson.build was used as a reference for doing this.
I've created another var null_dep to make things more clear.